ccc-js/src/main/javascript/package-res/lib/protovis.js

```diff
@@ -4601,17 +4603,24 @@ DateComponent.prototype.withPrecision = function(value) {
 };
 
 DateComponent.prototype.ticks = function(min, max, mult, options) {
-  var ticks = [],
-      tick  = new Date(min);
-
-  // TODO: need both?
-  // -- Floor start date (floor, independently of roundInside).
-  this.floor(tick, options);
-  // Floor the start date to the chosen multiple's first date.
-  if(mult > 1) this.floorMultiple(tick, mult, options);
+  var ticks = [];
+  var tick = new Date(min);
+  var isRoundInside = pv.get(options, 'roundInside', 1);
+  var isPreserveLastTick = pv.get(options, 'preserveLastTick');
+
+  if(isPreserveLastTick) {
+    this.setTickStart(tick, mult);
+    tick = new Date(this.tickStart);
+  } else {
+    // TODO: need both?
+    // Floor start date (floor, independently of roundInside).
+    this.floor(tick, options);
+    // Floor the start date to the chosen multiple's first date.
+    if(mult > 1) this.floorMultiple(tick, mult, options);
+  } 
```

Comment:
As we'd discussed, it seems this logic could be accomplished by some simple modifications of the `floorMultiple` method, so I don't see why we need the `setTickStart` method.